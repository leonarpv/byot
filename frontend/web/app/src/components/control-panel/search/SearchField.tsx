import * as React from 'react';
import {Typography} from '@material-ui/core';
import {useTranslation} from 'react-i18next';
import {Input} from '@byot-frontend/web-common/src/components/elementary/form/Input';

interface Props {
  onChange: (value: string) => void;
  value: string;
  debounce?: number;
}

export const SearchField: React.FC<Props> = (props: Props) => {
  const {t} = useTranslation();
  return (
    <>
      <Typography variant="h6">{t('Search')}</Typography>
      <Input
        size="small"
        label={t('Type to search')}
        value={props.value}
        onChangeText={props.onChange}
        debounce={typeof props.debounce == 'undefined' ? 300 : props.debounce}
      />
    </>
  );
};
