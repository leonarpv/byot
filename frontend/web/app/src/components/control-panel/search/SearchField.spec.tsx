import * as React from 'react';
import {render, fireEvent} from '@testing-library/react';
import {SearchField} from './SearchField';

describe('<SearchField/>', () => {
  it('should render and trigger the event', () => {
    const onChange = jest.fn();
    const {container} = render(<SearchField onChange={onChange} value="" debounce={0} />);
    const input = container.querySelector('input');
    expect(input).not.toBeNull();
    fireEvent.change(input!, {target: {value: 'something'}});
    expect(onChange).toBeCalled();
    expect(container).toMatchSnapshot();
  });
});
