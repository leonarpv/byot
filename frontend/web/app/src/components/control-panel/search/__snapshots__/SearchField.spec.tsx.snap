// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchField/> should render and trigger the event 1`] = `
<div>
  <h6
    class="MuiTypography-root MuiTypography-h6"
  >
    Search
  </h6>
  <div
    class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
    debounce="0"
  >
    <label
      class="MuiFormLabel-root MuiInputLabel-root makeStyles-label-2 MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-marginDense MuiInputLabel-filled"
      data-shrink="false"
    >
      Type to search
    </label>
    <div
      class="MuiInputBase-root MuiFilledInput-root makeStyles-filledRoot-1 makeStyles-filledRoot-3 MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-marginDense MuiFilledInput-marginDense"
    >
      <input
        aria-invalid="false"
        class="MuiInputBase-input MuiFilledInput-input MuiInputBase-inputMarginDense MuiFilledInput-inputMarginDense"
        type="text"
        value=""
      />
    </div>
  </div>
  <div
    class="MuiCollapse-container MuiCollapse-hidden"
    style="min-height: 0px;"
  >
    <div
      class="MuiCollapse-wrapper"
    >
      <div
        class="MuiCollapse-wrapperInner"
      >
        <p
          class="MuiFormHelperText-root MuiFormHelperText-contained"
          data-testid="common-elementary-form-input-helperText"
        >
          <span>
            â€‹
          </span>
        </p>
      </div>
    </div>
  </div>
</div>
`;
